core/src/main/kotlin/com/code42/jenkins/pipelinekt/core/secrets/VaultSecret.kt

```diff
@@ -0,0 +1,7 @@
+package com.code42.jenkins.pipelinekt.core.secrets
+
+data class VaultSecret(val envVar: String, val vaultKey: String) {
```

Comment:
Not mission critical but we dont seem to have docs for this class but we do for the WithVault one below. Would be nice to have all the methods/classes but I would be ok with that being its own body of work.

Reply:
I added in javadoc.